File: projects/09/ruby/temperature_tester.rb
# Test the temperature.rb and has the main function
# Author: David Yoo (gy24)
# Date: 03/22/2022

require './temperature'

def main
  puts "Temperature examples:"
  puts "\t20.5 C"
  puts "\t10 K"
  puts "\t35.1 F"
  print "Please enter a start temperature: "
  start, error = Temperatures::Temperature.read_temp
  while error
    print "Invalid input, please enter a valid input: "
    start, error = Temperatures::Temperature.read_temp
  end

  print " Please enter a stop temperature: "
  stop, error = Temperatures::Temperature.read_temp
  while error
    print "Invalid input, please enter a valid input: "
    stop, error = Temperatures::Temperature.read_temp
  end
  
  # NEED TO FIX THE PRINT MESSAGE, AND VARIABLE NAME
  print "Please enter a step amount of degrees: "
  step_string = gets.chomp
  step = 0.0
  while not Temperatures::Temperature.is_numeric(step_string)
    print "Invalid input, please enter a valid input: "
    step_string = gets.chomp
  end
  step = step_string.to_f

  if start.equals(stop)
    puts("Start and stop temperatures are equal.")
    return
  end
  
  current = start
  while true
    if start.less_than(stop) != current.less_than(stop) && !current.equals(stop)
      break
    end

    current.toF.print_temp
    print "\t\t"
    current.toC.print_temp
    print "\t\t"
    current.toK.print_temp
    puts ""

    if start.less_than(stop)
      current = current.raise(step)
    else
      current = current.lower(step)
    end
  end
end

main

File: projects/09/java/TemperatureTester.java
/*
 * Tester for Temperature.java, and it also contains main function
 * Author: David Yoo (gy24)
 * Date: 03/23/2020
 */
import java.util.Scanner;
class TemperatureTester {
    public static void main(String[] args) {
	System.out.println("Temperature examples:");
	System.out.println("\t20.5 C");
	System.out.println("\t10 K");
	System.out.println("\t35.1 F");
	System.out.print("Please enter a start temperature: ");
	Temperature startTemp = Temperature.readTemp();
	while (startTemp == null) {
	    System.out.print("Invalid input, please enter a valid input: ");
	    startTemp = Temperature.readTemp();
	}
	System.out.print("Please enter a stop temperature: ");
	Temperature stopTemp = Temperature.readTemp();
	while (stopTemp == null) {
	    System.out.print("Invalid input, please enter a valid input: ");
	    stopTemp = Temperature.readTemp();
	}
        
	Scanner scanner = new Scanner(System.in);
	String stepString = "";
	float amount = 0;
	boolean hasStep = false;
	int i = 0;
	while (!hasStep) {
	    if (i == 0)
			System.out.print("Please enter a step amount of degrees: ");
	    else
			System.out.print("Invalid input, please enter a valid input: ");
	    try {
			stepString = scanner.nextLine();
			amount = Float.parseFloat(stepString);
			hasStep = true;
	    } catch (NumberFormatException e) {	}
	    i++;
	}
	scanner.close();

	if (startTemp.equals(stopTemp)) {
	    System.out.println("Start and stop temperatures are equal.");
	    return;
	}

	Temperature current = startTemp;
	while (true) {
	    if (startTemp.lessThan(stopTemp) != current.lessThan(stopTemp) && !current.equals(stopTemp))
		break;

	    current.toF().print();
	    System.out.print("\t\t");
	    current.toC().print();
	    System.out.print("\t\t");
	    current.toK().print();
	    System.out.println();

	    if (startTemp.lessThan(stopTemp))
			current = current.raise(amount);
	    else
			current = current.lower(amount);
	}
	}	
}